components/automate-cli/cmd/chef-automate/verify.go

```diff
@@ -0,0 +1,181 @@
+package main
+
+import (
+	"github.com/chef/automate/components/automate-cli/pkg/docs"
+	"github.com/chef/automate/components/automate-cli/pkg/status"
+	verification "github.com/chef/automate/lib/verification"
+	"github.com/spf13/cobra"
+)
+
+var verifyCmd = &cobra.Command{
+	Use:   "verify",
+	Short: "Verify Chef Automate setup",
+	Long:  "Verify Chef Automate config files and Infrastructre",
+	Annotations: map[string]string{
+		docs.Compatibility: docs.Compatibility,
+	},
+	Args: cobra.RangeArgs(0, 1),
+	RunE: runVerifyCmd,
+}
+
+var verifyCmdFlags = struct {
+	file                      string
+	haAWSProvision            bool
+	haAWSManagedProvision     bool
+	haOnpremDeploy            bool
+	haOnPremAWSManagedDeploy  bool
+	haOnPremCustManagedDeploy bool
+	haAWSDeploy               bool
+	haAWSManagedDeploy        bool
+	standaloneDeploy          bool
+	certificates              bool
+}{}
+
+func init() {
+	// flags for Verify Command
+	verifyCmd.PersistentFlags().BoolVar(
+		&verifyCmdFlags.haAWSProvision,
+		"ha-aws-provision",
+		false,
+		"Verifies the AWS Provision config")
+	verifyCmd.PersistentFlags().BoolVar(
+		&verifyCmdFlags.haAWSManagedProvision,
+		"ha-aws-managed-provision",
+		false,
+		"Verifies the AWS provision config with Managed services")
+	verifyCmd.PersistentFlags().BoolVar(
+		&verifyCmdFlags.haOnpremDeploy,
+		"ha-onprem-deploy",
+		false,
+		"Verifies the On-Prem setup and config with Chef Managed services")
+	verifyCmd.PersistentFlags().BoolVar(
+		&verifyCmdFlags.haOnPremAWSManagedDeploy,
+		"ha-onprem-aws-deploy",
+		false,
+		"Verifies the On-Prem setup and config with AWS Managed services")
+	verifyCmd.PersistentFlags().BoolVar(
+		&verifyCmdFlags.haOnPremCustManagedDeploy,
+		"ha-onprem-customer-deploy",
+		false,
+		"Verifies the On-Prem setup and config with Customer Managed services")
+	verifyCmd.PersistentFlags().BoolVar(
+		&verifyCmdFlags.haAWSDeploy,
+		"ha-aws-deploy",
+		false,
+		"Verifies the AWS Deployment setup and config")
+	verifyCmd.PersistentFlags().BoolVar(
+		&verifyCmdFlags.haAWSManagedDeploy,
+		"ha-aws-managed-deploy",
+		false,
+		"Verifies the AWS Deployement setup and config with Managed Services")
+	verifyCmd.PersistentFlags().BoolVar(
+		&verifyCmdFlags.standaloneDeploy,
+		"standalone-deploy",
+		false,
+		"Verifies the Automate standalone setup and config")
+	verifyCmd.PersistentFlags().BoolVar(
+		&verifyCmdFlags.certificates,
+		"certificates",
+		false,
+		"Verifies the certificates provided in the file")
+	verifyCmd.PersistentFlags().StringVar(
+		&verifyCmdFlags.file,
+		"file",
+		"",
+		"Config file that needs to be verified")
+	RootCmd.AddCommand(verifyCmd)
+}
+
+func runVerifyCmd(cmd *cobra.Command, args []string) error {
```

Comment:
Can we break this function into multiple smaller function? Cognitive Complexity is high here.

Reply:
Done